sdv/constraints/tabular.py

```diff
@@ -163,8 +168,25 @@ def _transform(self, table_data):
                 Transformed data.
         """
         lists_series = pd.Series(table_data[self._columns].values.tolist())
+
+        non_string_cols = [dt for x, dt in table_data.dtypes[
```

Comment:
Searching for `object` may not be enough, since this does not guarantee that they are strings.

On the other side, the uuids need to be created at `fit` time, not `transform`, since we want the same combinations to map to the same uuid at every transform call.

Finally, what about just assuming that everything is not a string?

I liked how you used the `tuple`, so everything could be made much simpler by just doing (without even caring about whether values are strings or not):

```
# fit
combinations = table_data[self._columns].drop_duplicates()
self._combinations_to_uuids = {}
self._uuids_to_combinations = {}
for combination in self._combinations.itertuples(index=False, name=None):
    uuid = str(uuid4())
    self._combinations_to_uuids[combination] = uuid
    self._uuids_to_combinations[uuid] = combination

# transform
combinations = table_data[columns].itertuples(index=False, name=None)
uuids = map(_combinations_to_uuids.get, combinations)
table_data[self._joint_column] = list(uuids)
return table_data.drop(self._columns, axis=1)

# reverse transform
table_data[self._columns] = table_data.pop(self._joint_column).map(self._uuids_to_combinations)
```

By doing this, we can even skip checking whether the separator is in the data, because it does not matter any more.